File: src/data/Models/MatchEvent.cs
using ElevenLegends.Data.Enums;

namespace ElevenLegends.Data.Models;

/// <summary>
/// A single event that occurred during a match tick.
/// </summary>
public sealed record MatchEvent
{
    public required int Tick { get; init; }
    public required EventType Type { get; init; }

    /// <summary>The player who performed the action.</summary>
    public required int PlayerId { get; init; }

    /// <summary>Optional secondary player (e.g., assist provider, fouled player).</summary>
    public int? SecondaryPlayerId { get; init; }

    /// <summary>Human-readable description of the event.</summary>
    public string Description { get; init; } = "";

    /// <summary>Rating impact for the primary player.</summary>
    public float RatingImpact { get; init; }
}

File: engine/models/match-event.ts
import { EventType } from "@/engine/enums";

/** A single event that occurred during a match tick. */
export interface MatchEvent {
  tick: number;
  type: EventType;

  /** The player who performed the action. */
  playerId: number;

  /** Optional secondary player (e.g., assist provider, fouled player). */
  secondaryPlayerId: number | null;

  /** Human-readable description of the event. */
  description: string;

  /** Rating impact for the primary player. */
  ratingImpact: number;
}
